src/hooks/useApi.ts

```diff
@@ -11,28 +12,15 @@ const BASE_URL = '/api';
  */
 const useApi = <T>(path: string): [Error | null, T | null] => {
   const [result, setResult] = useState<T | null>(null);
-  const [error, setError] = useState<Error | null>(null);
+  const callback = useCallback(async () => {
+    setResult(null);
+    setResult(await get(path));
+  }, [path]);
+  const [error, handler] = useErrorHandler(callback);
 
   useEffect(() => {
-    const sendRequest = async () => {
```

Comment:
This is the same as it was before, it's now just been split up in the `useErrorHandler()` hook and the `get()` helper function from `api.service.ts`. This is to facilitate reuseability of the different parts.